src/lib/settings/settings.py

```diff
@@ -280,6 +280,32 @@ def get_metrics_collection_interval_min(self) -> int:
         return self.processed_settings[SettingFileNames.GENERAL]["tooling_environment"][
             "metrics_collection_interval_min"
         ]
+    
+    def get_grafana_mandatory_settings(self) -> tuple[str, str]:
+        """
+        Get mandatory grafana settings:
+            - grafana_vpc_id: VPC ID for Grafana instance
+            - grafana_security_group_id: Security Group ID for Grafana instance
+        If not provided, the Grafana stack will be skipped.
+        """
+        grafana_settings = self.processed_settings[SettingFileNames.GENERAL]["tooling_environment"]
+        return (
+            grafana_settings.get("grafana_vpc_id"),
+            grafana_settings.get("grafana_security_group_id"))
+
+    def get_grafana_optional_settings(self) -> tuple[str, str, str]:
```

Comment:
I'm not sure we need to separate mandatory and optional settings into a different methods.
Maybe, we can just have get_grafana_settings?

Reply:
combined into one method